src/transport/SecureSessionMgr.cpp

```diff
@@ -187,7 +187,8 @@ CHIP_ERROR SecureSessionMgr::SendMessage(SecureSessionHandle session, PayloadHea
     // Retain the packet buffer in case it's needed for retransmissions.
     if (bufferRetainSlot != nullptr)
     {
-        encryptedMsg = msgBuf.Retain();
+        encryptedMsg        = msgBuf.Retain();
```

Comment:
why do we save the encryptedMsg first? Why not:

```cpp
*bufferRetainSlot = msgBuf.Retain();
```

that should make it clear that encryptedMsg does not contain a valid value here. We could make it even `encryptedMsg = nullptr` if we want to make it even more obvious.

Reply:
The original purpose is to hold the reference to encrypted message in stack variable. In case of any failures, the reference is not returned, and this stack variable will automatically free the reference on returning from the function.

If we use *bufferRetainSlot to hold the retained buffer directly, we don't need encryptedMsg at all.